src/buildHTML.js

```diff
@@ -575,12 +579,48 @@ var groupTypes = {
             totalHeight += depth + gap; // \@yargarraycr
             body[r] = outrow;
         }
+
         var offset = totalHeight / 2 + fontMetrics.metrics.axisHeight;
         var coldescriptions = group.value.cols || [];
         var cols = [];
         var colsep;
-        for (c = 0; c < nc; ++c) {
-            var coldescr = coldescriptions[c] || {};
+        var colDescrNum;
+        for (c = 0, colDescrNum = 0;
+             c < nc || colDescrNum < coldescriptions.length;
+             ++c, ++colDescrNum) {
+            var coldescr = coldescriptions[colDescrNum] || {};
+
+            var lastWasSeparator = false;
+            while (coldescr.type === "separator") {
+                if (lastWasSeparator) {
```

Comment:
Since we're only going to have separators within this while loop, it'd probably be easier to grok if `lastWasSeparator` was renamed to `firstSeparator` and set it to `true` initially and then check `if (!firstSeparator) { // add a gap }`.